.github/scripts/ensure_actions_will_cancel.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+import argparse
+import sys
+import yaml
+import io
+import re
+
+from pathlib import Path
+
+
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+WORKFLOWS = REPO_ROOT / ".github" / "workflows"
+
+
+def concurrency_key(filename):
+    workflow_name = filename.with_suffix("").name.replace("_", "-")
+    return f"{workflow_name}-${{{{ github.event.pull_request.number || github.sha }}}}"
+
+
+SKIP = {
+    "update_disabled_tests.yml",
+    "stale_pull_requests.yml",
+    "update_s3_htmls.yml",
+    "push_nightly_docker_ghcr.yml",
+}
+
+
+def should_skip(filename):
+    if filename.name in SKIP:
+        return True
+
+    with open(filename, "r") as f:
+        content = f.read()
+        if "@generated DO NOT EDIT MANUALLY" in content:
+            return True
+
+    data = yaml.safe_load(content)
+    on = str(data.get("on", data.get(True, None)))
+    if "'pull_request'" not in on and "'workflow_dispatch'" not in on:
+        return True
+
+    return False
+
+
+def insert_concurrency_keys(filename):
+    # Import here since this isn't installed in CI where this script runs
+    from ruamel.yaml import YAML
```

Comment:
instead of doing this, why not just install this lib in CI? that's what we were doing before

Reply:
This doesn't need to run in CI at all since it's just a one-off tool to add the `concurrency:` entry everywhere, so we'd be installing it to import it and not do anything with it

Reply:
hm I suppose that's true with the current approach; I guess this would be necessary for the alternate approach I ask about in my comment below, which could either be a reason to install the lib or a reason to keep the current approach, depending on your perspective